src/datasources/balances-api/zerion-balances-api.service.ts

```diff
@@ -168,4 +236,62 @@ export class ZerionBalancesApi implements IBalancesApi {
       );
     return chainName;
   }
+
+  private _mapCollectibles(
+    zerionCollectibles: ZerionCollectible[],
+  ): Collectible[] {
+    return zerionCollectibles.map(
+      ({ attributes: { nft_info, collection_info } }) => ({
+        address: nft_info.contract_address,
+        tokenName: nft_info.name ?? '',
+        tokenSymbol: nft_info.name ?? '',
+        logoUri: collection_info?.content?.icon.url ?? '',
+        id: nft_info.token_id,
+        uri: nft_info.content?.detail?.url ?? null,
+        name: collection_info?.name ?? null,
+        description: collection_info?.description ?? null,
+        imageUri: nft_info.content?.preview?.url ?? '',
+        metadata: nft_info.content,
+      }),
+    );
+  }
+
+  /**
+   * Zerion represents cursor offsets by base64 string
+   * contained within double quotation marks.
+   *
+   * @param offset number representing the offset
+   * @returns base64 string representing the offset
+   */
+  private _encodeZerionPageOffset(offset?: number): string | null {
+    return offset
+      ? Buffer.from(`"${offset}"`, 'utf8').toString('base64')
+      : null;
+  }
+
+  /**
+   * Zerion uses page[size] as pagination limit.
+   * Zerion uses page[after] as pagination offset, which is a
+   * base64 string contained within double quotation marks.
+   *
+   * @param url Zerion-formatted string representing an URL
+   * @returns URL string optionally containing "limit" and "offset" query params
+   */
+  private _decodeZerionPagination(url: string): string {
+    const zerionUrl = new URL(url);
+    const size = zerionUrl.searchParams.get('page[size]');
+    const after = zerionUrl.searchParams.get('page[after]');
+
+    const resultUrl = new URL(zerionUrl);
```

Comment:
We need not instantiated `resultUrl` here - we can use `zerionUrl`.

Reply:
Good catch as well! [1a730ce](https://github.com/safe-global/safe-client-gateway/pull/1121/commits/1a730ce238b8101cd84728a16c784176c139b255)